com.kms.cura.dal/src/com/kms/cura/dal/database/DoctorUserDatabaseHelper.java

```diff
@@ -154,6 +151,32 @@ protected void insertToDoctorReferenceTables(DoctorUserEntity entity) throws SQL
 		insertReferenceRowsToReferenceTable(Doctor_SpecialityColumn.TABLE_NAME, referenceRows);
 	}
 
+	protected void insertWorkingHour(DoctorUserEntity doctorUserEntity) throws SQLException {
+		List<OpeningHour> hours = doctorUserEntity.getFacility().get(0).getOpeningHours();
+		PreparedStatement stmt = null;
+		StringBuilder builder = new StringBuilder();
+		String doctorID = doctorUserEntity.getId();
+		String facilityID = doctorUserEntity.getFacility().get(0).getId();
+		builder.append("INSERT INTO ");
+		builder.append(Doctor_FacilityColumn.TABLE_NAME);
+		builder.append(" VALUES (?, ?, ?, ?, ?);");
+		try{
+			for(int i=0; i<hours.size();++i){
+				OpeningHour hour = hours.get(i);
+				stmt = con.prepareStatement(builder.toString());
+				stmt.setInt(1, Integer.parseInt(doctorID));
+				stmt.setInt(2, Integer.parseInt(facilityID));
+				stmt.setInt(3, hour.getDayOfTheWeek().getCode());
+				stmt.setTime(4, hour.getOpenTime());
+				stmt.setTime(5, hour.getCloseTime());
+				stmt.executeUpdate();
```

Comment:
except from GET methods, all other methods that changing database need to rollback if exceptions happened